### PR TITLE
Move containers from Docker Hub to ACR

### PR DESCRIPTION
**Pull request by danewalton**
- Moves images from Docker Hub to Azure Container Registry
- Reshuffles tests to group together "like" tests

### CODE REVIEW

**Review by jspaith — APPROVED** [review]

File: build/.vsts-ci.yml [hunk 1]
```diff
@@ -285,11 +322,10 @@ jobs:
...
-  variables:
-    _PREVIEW_VSTS_DOCKER_IMAGE: "aziotbld/vsts-c-wolfssl"
+  container: linux-c-ubuntu-wolfssl
   pool: 
     name: 'sdk-c--ubuntu-18'
-  displayName: 'wolfssl'
+  displayName: 'WolfSSL'
   steps:
   - script: |
      if [ -f "jenkins/linux_wolfssl.sh" ]
```
[the remark this answers is not in the document]
Comment (danewalton) [reply build/.vsts-ci.yml:331]: good callout I'll remove it

[the remark this answers is not in the document]
Comment (danewalton) [reply build/.vsts-ci.yml]: actually let me see if I can add the test. I'll try and find the script to do it.

[the remark this answers is not in the document]
Comment (danewalton) [reply build/.vsts-ci.yml:331]: Working now (no e2e but compiles with wolfssl)

**Pull request merged**